core/src/main/java/hudson/model/Computer.java

```diff
@@ -195,6 +195,8 @@
 
     protected final Object statusChangeLock = new Object();
 
+    private final Object logDirLock = new Object();
```

Comment:
```suggestion
    private final Object logDirLock = new String("logDirLock);
```

since a future version of Java may not support synchronizing on `Object`

Reply:
Ah? Didn't know. Any reference?

Reply:
At least that is my reading of: http://cr.openjdk.java.net/~briangoetz/valhalla/sov/02-object-model.html

Reply:
As I understand it would depend on whether the `Object` class is treated as an inline class or not.

Reply:
I thought I had seen some explicit discussion about this idiom, whereby code would need to be migrated to using a `Lock`, a better monitor, or at least

```java
final Object lock = Object.newIdentityObject();
// ..
synchronized (lock) {
    // …
}
```

but maybe I am reading more into it than was really there.